katsdpingest/ingest_session.py

```diff
@@ -612,27 +620,58 @@ def _zero_counters(self) -> None:
         if not len(self.bls_ordering.sdp_bls_ordering):
             raise ValueError('No baselines (bls_ordering = {}, antenna_mask = {})'.format(
                 self.cbf_attr['bls_ordering'], antenna_mask))
 
-        # Determine which channel mask to use for each baseline
-        thresholds = np.array(self.telstate_cbf.get('channel_mask_max_baseline_lengths', []))
-        bls = self.bls_ordering.sdp_bls_ordering
-        antenna_names = set(input_name[:-1] for input_name in bls.flat)
-        antennas = [katpoint.Antenna(self.telstate['{}_observer'.format(name)])
-                    for name in antenna_names]
-        ref = antennas[0].array_reference_antenna()
-        # Turn each antenna into a location East/North/Up of the reference
-        locations = {antenna.name: np.array(ref.baseline_toward(antenna))
-                     for antenna in antennas}
-        lengths = np.array([np.linalg.norm(locations[a[:-1]] - locations[b[:-1]])
-                            for (a, b) in bls])
-        self.bls_channel_mask_idx = np.searchsorted(thresholds, lengths, 'right').astype(np.uint32)
-        self.n_channel_masks = len(thresholds) + 1
+        # Pre-compute channel masks from the RFI model
+        if self._rfi_mask_model is not None:
+            bls = self.bls_ordering.sdp_bls_ordering
+            antenna_names = set(input_name[:-1] for input_name in bls.flat)
+            antennas = [katpoint.Antenna(self.telstate['{}_observer'.format(name)])
+                        for name in antenna_names]
+            ref = antennas[0].array_reference_antenna()
+            # Turn each antenna into a location East/North/Up of the reference
+            locations = {antenna.name: np.array(ref.baseline_toward(antenna))
+                         for antenna in antennas}
+            lengths = np.array([np.linalg.norm(locations[a[:-1]] - locations[b[:-1]])
+                                for (a, b) in bls]) << u.m
+            cbf_spw = SpectralWindow(
+                self.cbf_attr['center_freq'], None, len(self.channel_ranges.cbf),
+                bandwidth=self.cbf_attr['bandwidth'], sideband=1)
+            spw = cbf_spw.subrange(self.channel_ranges.input.start,
+                                   self.channel_ranges.input.stop)
+            freqs = spw.channel_freqs << u.Hz
+            # Get masks as a baseline x freq array
+            masks = self._rfi_mask_model.is_masked(freqs[np.newaxis, :], lengths[:, np.newaxis])
+            # Typically the model will only have a few threshold lengths, so
+            # most rows will be the same. Reduce it to just unique rows, with
+            # a lookup table.
+            masks, indices = np.unique(masks, axis=0, return_inverse=True)
+            self.bls_channel_mask_idx = indices.astype(np.uint32)
+            self.static_masks = masks * np.uint8(STATIC)
+        else:
+            self.bls_channel_mask_idx = np.zeros(len(self.bls_ordering.sdp_bls_ordering), np.uint32)
```

Comment:
I think we should log this fallback as a warning.

Reply:
`_init_models` already does logging if it couldn't initialise an RFI model. Should I upgrade that info to a warning?

Reply:
I've upgraded the message in `_init_models` to a warning.